### PR TITLE
feature(js): elgg/spinner now supports optional text to be displayed

### LINKED ISSUE #8979: FR: elgg_spinner support text to show in spinner
When starting the Elgg spinner (loading ajax content), it would be nice if we optionally could add text to the `spinner.start()` function so we can provide feedback to the end user what we are waiting on.

So `spinner.start(elgg.echo('Waiting on the pizza guy'));` results in:
waiting on the pizza guy

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #8979

### CODE REVIEW

File: views/default/elgg/spinner.js [hunk 1]
```diff
@@ -4,21 +4,34 @@ define(function (require) {
...
 
 		stop: function () {
 			active = false;
-			$('body').removeClass('elgg-spinner-active');
-		}
+			$('body').removeClass('elgg-spinner-active').find();
+		},
+		
+		setText: function (text) {
+			$('.elgg-spinner .elgg-spinner-text').html(text);
```
Comment (mrclay) [inline views/default/elgg/spinner.js]: For XSS mitigation, either use text() to set the text, or rename the argument HTML so users are clear that they need to escape manually.

**Comment by hypeJunction** [comment]
I am not crazy about this. We would have to figure out a way to make it work with `elgg/Ajax`. Plugins shouldn't even need to require spinner anymore.

File: views/default/elgg/spinner.js [hunk 2]
```diff
@@ -4,21 +4,34 @@ define(function (require) {
...
 			}, SHOW_DELAY);
+			
+			this.setText(text);
 		},
 
 		stop: function () {
 			active = false;
-			$('body').removeClass('elgg-spinner-active');
-		}
+			$('body').removeClass('elgg-spinner-active').find();
```
Comment (mrclay) [inline views/default/elgg/spinner.js]: Why the find()?

**Comment by mrclay** [comment]
Right, elgg/Ajax fetches would wipe out the text (good or bad?). For typical ajax stuff it'd disappear before you could read it anyway. I could imagine using this only on a very slow process, but I'd probably use a more prominent UI message in that case.

**Comment by mrclay** [comment]
If @jdalsem can address my comments I'm cool with it though.

**Pull request merged**